src/BBT.MaybePattern/Maybe.cs

```diff
@@ -98,6 +98,23 @@ public Maybe<TResult> Some<TResult>(Func<T, TResult> func)
             return maybe;
         }
 
+        /// <summary>
+        /// Gets the <see cref="Maybe{TResult}"/> of the projection function <paramref name="func"/>.
+        /// </summary>
+        /// <typeparam name="TResult">The type of the projected result.</typeparam>
+        /// <param name="func">The projection function.</param>
+        /// <returns>The projected maybe.</returns>
+        public Maybe<TResult> Some<TResult>(Func<T, Maybe<TResult>> func)
+            where TResult : class
```

Comment:
Shouldn't we cover the case where `TResult` is a struct type as well?
I fancy the idea of keeping Maybe for `class` and `struct` types at feature parity.

Reply:
Nevermind, just seen that the `MaybeStruct` already offers that.

Reply:
no, you're right with your assumption. I added the missing three methods / overloads. Because of method renaming this issue became breaking now.